common/determined_common/experimental/checkpoint/_checkpoint.py

```diff
@@ -23,11 +23,36 @@ class CheckpointState(enum.Enum):
 
 class Checkpoint(object):
     """
-    Class representing a checkpoint. Contains methods for downloading
-    checkpoints to local storage and loading checkpoints into memory.
+    A ``Checkpoint`` represents a trained model.
+
+    This class provides helper functionality for downloading checkpoints to
+    local storage and loading checkpoints into memory.
 
     The :class:`~determined.experimental.TrialReference` class contains methods
     that return instances of this class.
+
+    Arguments:
```

Comment:
question: I am curious as to what is the motivation for publicly documenting these `args`? do we expect users to be creating their own instances of the `Checkpoint` object?

Reply:
Not really; the goal here is to document the public fields of the `Checkpoint` object (which happen to be the same as the arguments to the constructor). We use "Arguments" in the same way for other objects like `Model`. If you use the "Attributes" syntax, the docs that get generated are a lot more verbose. Maybe worth switching over in the future? But we should make it consistent.

Reply:
My concern here is that it may be confusing to the users to expose these Arguments to the users since we never expect them to instantiate this object themselves, we just end up cluttering their docs with (mostly) useless information.

Reply:
Yep -- makes sense. I'll see if there's a better way to do this in the future.